server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettings.java

```diff
@@ -367,128 +298,68 @@ public TimeValue getRerouteInterval() {
         return rerouteInterval;
     }
 
-    String describeLowThreshold() {
-        return freeBytesThresholdLow.equals(ByteSizeValue.ZERO)
-            ? Strings.format1Decimals(100.0 - freeDiskThresholdLow, "%")
-            : freeBytesThresholdLow.toString();
-    }
-
-    String describeHighThreshold() {
-        return freeBytesThresholdHigh.equals(ByteSizeValue.ZERO)
-            ? Strings.format1Decimals(100.0 - freeDiskThresholdHigh, "%")
-            : freeBytesThresholdHigh.toString();
-    }
-
-    String describeFloodStageThreshold() {
-        return freeBytesThresholdFloodStage.equals(ByteSizeValue.ZERO)
-            ? Strings.format1Decimals(100.0 - freeDiskThresholdFloodStage, "%")
-            : freeBytesThresholdFloodStage.toString();
-    }
-
-    String describeFrozenFloodStageThreshold(ByteSizeValue total) {
-        ByteSizeValue maxHeadroom = this.frozenFloodStageMaxHeadroom;
-        RelativeByteSizeValue floodStage = this.frozenFloodStage;
-        if (floodStage.isAbsolute()) {
-            return floodStage.getStringRep();
-        } else if (floodStage.calculateValue(total, maxHeadroom).equals(floodStage.calculateValue(total, null))) {
-            return Strings.format1Decimals(floodStage.getRatio().getAsPercent(), "%");
+    private String describeThreshold(
+        ByteSizeValue total,
+        RelativeByteSizeValue watermark,
+        ByteSizeValue maxHeadroom,
+        boolean includeSettingKey,
+        String watermarkSettingKey,
+        String maxHeadroomSettingKey
+    ) {
+        if (watermark.isAbsolute()) {
+            return includeSettingKey ? watermarkSettingKey + "=" + watermark.getStringRep() : watermark.getStringRep();
+        } else if (watermark.calculateValue(total, maxHeadroom).equals(watermark.calculateValue(total, null))) {
+            String value = watermark.getStringRep();
+            return includeSettingKey ? watermarkSettingKey + "=" + value : value;
         } else {
-            return "max_headroom=" + maxHeadroom;
+            return includeSettingKey
+                ? maxHeadroomSettingKey + "=" + maxHeadroom.getStringRep()
+                : "max_headroom=" + maxHeadroom.getStringRep();
```

Comment:
I don't think this case is covered by `DiskThresholdSettingsTests#doTestDescriptions` - could you add a few lines of test code about this branch there?

Reply:
Hi, yes it's covered in there if you see:

```
        String frozenFloodMaxHeadroomPrefix = includeKey
            ? DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_FROZEN_MAX_HEADROOM_SETTING.getKey() + "="
            : "max_headroom=";
```

It only applies to the frozen tier max headroom, and there is a relevant test:

```
        assertThat(
            diskThresholdSettings.describeFrozenFloodStageThreshold(thousandTb, includeKey),
            equalTo(frozenFloodMaxHeadroomPrefix + "10gb")
        );
```

Please unresolve this conversation if it doesn't suffice.

Reply:
Sorry I see now, don't mind me. 😕